code/datums/abilities/critter/stomach_retreat.dm

```diff
@@ -0,0 +1,61 @@
+/datum/targetable/critter/stomach_retreat
+	name = "Retreat to Stomach"
+	desc = "Turn yourself inside out for shelter! Must be inside a disposal chute."
+	icon_state = "mimic_stomach_retreat"
+	cooldown = 120 SECONDS
+	cooldown_after_action = TRUE
+	needs_turf = FALSE
+	var/inside = FALSE
+	var/obj/machinery/disposal/current_chute = null
```

Comment:
Unsure of feasibility but could this use `COMSIG_ATOM_ENTERED` instead? That way it could be used on an arbitrary atom, and a whitelist could live here? Then they could be inside crates... or lockers...